### PR TITLE
+Add sample for MFC

### PR DESCRIPTION
**Pull request by milanac030988**

### CODE REVIEW

**Comment by test-fullautomation** [comment]
Hi Cuong,
very good!
Please check my questions.
Thank you,
 Thomas

File: MFCSample/MicroserviceManager/MicroserviceManagerDlg.cpp [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+	return resultMap;
+}
+
+void CMicroserviceManagerDlg::SetSwitchRequest(string selectedDevice, int switchNum, string state)
+{
+	ServiceClient* srv = new ServiceClient("ServiceCleware", "127.0.0.1", 5672, "ServiceClewareKey");
+	Json::Value jsonObject;
+	jsonObject[METHOD] = REQ_SET_SWITCH;
+
+	Json::Value listArray;
```
Comment (test-fullautomation) [inline MFCSample/MicroserviceManager/MicroserviceManagerDlg.cpp]: Hi Cuong,
is there no easier way to create an anonymous data structure on the fly directly in the interface of RequestService?
Thank you,
 Thomas

File: MFCSample/MicroserviceManager/ServiceClient.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    for (const auto& byte : bytes) {
+        ss << std::setw(2) << static_cast<int>(byte);
+    }
+
+    return ss.str();
+}
+
+int ServiceClient::RequestService(std::string jsonData, std::function<void(const std::string&)> callback) {
+    amqp_connection_state_t conn = amqp_new_connection();
+    amqp_socket_t* socket = amqp_tcp_socket_new(conn);
```
Comment (test-fullautomation) [inline MFCSample/MicroserviceManager/ServiceClient.cpp]: Hi Cuong,
for each Request a new Socket might be a performance issue?
Is it possible to use kind of socketpool here?
Thank you,
 Thomas

**Review by test-fullautomation — APPROVED** [review]
Hi Cuong,
please check my questions.
Thank you,
 Thomas

File: MFCSample/MicroserviceManager/ServiceClient.cpp (hunk 2 above)
Comment (milanac030988) [reply MFCSample/MicroserviceManager/ServiceClient.cpp]: Hi Thomas,
It's possible. I've just make the simple way for a running demo.
Will change in next update after some fixed bugs.
Thanks,
Cuong

File: MFCSample/MicroserviceManager/MicroserviceManagerDlg.cpp (hunk 1 above)
Comment (milanac030988) [reply MFCSample/MicroserviceManager/MicroserviceManagerDlg.cpp]: Hello Thomas,

I'm preparing a better design. 

I have many ideas about design, and I tried to implement them, but in the end, I got overwhelmed by a significant number of bugs. At the last minute, I decided to go for the simplest approach to create a running demo.

Thanks,
Cuong

Comment (test-fullautomation) [reply MFCSample/MicroserviceManager/MicroserviceManagerDlg.cpp]: Hi Cuong,
no problem. I wanted to give only my ideas. Nice, when you work already on both findings.

Best would be a as simple as possible transparent RPC call provided by generic functions. It came into my mind that maybe also macros could help here. I made long no c++ code :-)

Thank you,
 Thomas

**Comment by test-fullautomation** [comment]
Hi @milanac030988 ,
thank you for the update.
I will give this as intermediate version to Torsten and Andreas.
Thank you,
 Thomas

**Pull request merged**